src/test/java/uk/gov/digital/ho/proving/income/audit/AuditArchiveServiceIT.java

```diff
@@ -0,0 +1,203 @@
+            .andExpect(method(POST))
+            .andExpect(jsonPath("$.nino", is("nino_1")))
+            .andExpect(jsonPath("$.eventIds.*", containsInAnyOrder("corr-id-1")))
+            .andExpect(jsonPath("$.result", is("ERROR")))
+            .andExpect(jsonPath("$.resultDate", is("2019-01-02")))
+            .andRespond(withSuccess());
+
+        auditArchiveService.archiveAudit();
+    }
+
+    @Test
+    public void archiveAudit_errorThenNotFound_bestResultReturned() {
+        String errorRequest = fileUtils.buildRequest("corr-id-1", "2019-01-02 11:00:00.000", "nino_1");
+        String notFoundRequest = fileUtils.buildRequest("corr-id-2", "2019-01-03 12:00:00.000", "nino_1");
+        String notFoundResponse = fileUtils.buildResponseNotFound("corr-id-2", "2019-01-03 13:00:00.000");
+        String auditHistory = String.format("[%s, %s, %s]", errorRequest, notFoundRequest, notFoundResponse);
+        mockAuditService
+            .expect(requestTo(containsString("/history")))
+            .andExpect(method(POST))
+            .andRespond(withSuccess(auditHistory, APPLICATION_JSON));
+
+        mockAuditService
+            .expect(requestTo(containsString("/archive")))
+            .andExpect(method(POST))
+            .andExpect(jsonPath("$.nino", is("nino_1")))
+            .andExpect(jsonPath("$.eventIds.*", containsInAnyOrder("corr-id-1", "corr-id-2")))
+            .andExpect(jsonPath("$.result", is("NOTFOUND")))
+            .andRespond(withSuccess());
+
+        auditArchiveService.archiveAudit();
+    }
+
+    @Test
+    public void archiveAudit_multipleMatchingResults_latestResultReturned() {
+        String request1 = fileUtils.buildRequest("corr-id-1", "2019-01-01 12:00:00.000", "nino_1");
+        String response1 = fileUtils.buildResponse("corr-id-1", "2019-01-01 13:00:00.000", "nino_1", "true");
+        String request2 = fileUtils.buildRequest("corr-id-2", "2019-01-02 14:00:00.000", "nino_1");
+        String response2 = fileUtils.buildResponse("corr-id-2", "2019-01-02 15:00:00.000", "nino_1", "true");
+        String auditHistory = String.format("[%s, %s, %s, %s]", request1, response1, request2, response2);
+        mockAuditService
+            .expect(requestTo(containsString("/history")))
+            .andExpect(method(POST))
+            .andRespond(withSuccess(auditHistory, APPLICATION_JSON));
+
+        String endOfArchive = LocalDate.now().minusMonths(6).minusDays(1).format(DateTimeFormatter.ISO_DATE);
+        mockAuditService
+            .expect(requestTo(containsString("/archive")))
+            .andExpect(method(POST))
+            .andExpect(jsonPath("$.nino", is("nino_1")))
+            .andExpect(jsonPath("$.lastArchiveDate", is(endOfArchive)))
+            .andRespond(withSuccess());
+
+        auditArchiveService.archiveAudit();
+    }
+
+    @Test
+    public void archiveAudit_multipleMatchingAndMultipleresults_latestResultForAllRequestsReturned() {
```

Comment:
This test is too big for me to understand. Wouldn't 2 different Ninos be enough to confirm it works for multiple? And the PASS/FAIL/NOT_FOUND precedence can be done in other, smaller tests.

Reply:
Let me have a quick refactor and see what you think - if it's still too big then I'll have a rethink.